napari/components/overlays/scale_bar.py

```diff
@@ -54,3 +57,4 @@ class ScaleBarOverlay(CanvasOverlay):
         default_factory=lambda: ColorValue([0, 0, 0, 0.6])
     )
     unit: Optional[str] = None
+    length: Optional[int] = None
```

Comment:
Mypy found this:

```suggestion
    length: Optional[float] = None
```

Reply:
ah yeah with type hinting float entails int as well

Reply:
@brisvag 

I turned it into `int` because if set to `float`, it will show the number on the scale bar as `100.0`, which I found a bit strange for a scale bar ^^" Maybe I could just add an `int()` conversion in the length calculation as the scale bar will probably only be displayed in int values of pixels?

Reply:
For the dynamic scalebar it probably didn't pop up because the variable `_target_length` isn't tracked by mypy. One workaround could be to just force the value to integer upon display here?

```python
self.node.text.text = f'{int(new_dim):~}'
```

Reply:
I feel like this is the kind of place where if we say "of course no one will ever want to set a float scale bar", the next day we get an issue of someone who does xD

What about using something like:

```py
f"{length:g}"
```
for the rendered value? Should cut down to significant digits.

Reply:
Hold on, I just realized `pint` has some special formatting options...

Reply:
Got it, you need to change this:

```py
self.node.text.text = f'{new_dim:~}'
```

to 
```py
self.node.text.text = f'{new_dim:g~}'
```

Since we're here, I'd suggest to actually do:

```
self.node.text.text = f'{new_dim:g~#P}'
```

To get the significant digits + "short compact pretty" formatting: https://pint.readthedocs.io/en/stable/user/formatting.html#string-formatting-specification

Reply:
Changed in b17691643adc911a21ec278805d797e1da5f3253 and 2677f4d904f9dd51d149e3f2be7278cf6d631da6